src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java

```diff
@@ -17,66 +17,51 @@
 import static uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.Documents.insertNewRecords;
 
 @Service
-public class CallbackProcessor {
-    private static final Logger log = LoggerFactory.getLogger(CallbackProcessor.class);
+public class AttachCaseCallbackService {
+    private static final Logger log = LoggerFactory.getLogger(AttachCaseCallbackService.class);
     private final CcdApi ccdApi;
 
-    public CallbackProcessor(CcdApi ccdApi) {
+    public AttachCaseCallbackService(CcdApi ccdApi) {
         this.ccdApi = ccdApi;
     }
 
-    @SuppressWarnings("squid:S1135")
-    //TODO: RPE-822 cleanup
-    public List<String> process(String eventType, String eventId, CaseDetails caseDetails) {
+    public List<String> process(CaseDetails caseDetails) {
         return Validation
             .combine(
-                isAttachEvent(eventType),
-                isAttachToCaseEvent(eventId),
                 hasJurisdiction(caseDetails),
-                hasCaseTypeId(caseDetails),
                 hasCaseReference(caseDetails),
-                hasAScannedRecord(caseDetails),
-                hasCaseDetails(caseDetails)
+                hasAnId(caseDetails),
+                hasAScannedRecord(caseDetails)
             )
             .ap(this::attachCase)
             .getOrElseGet(Value::toJavaList);
     }
 
-    @SuppressWarnings({"squid:S1172", "squid:S1135", "squid:S1854"})
-    //TODO: RPE-822 these are for the validations of the incoming request and is a WIP
-    private List<String> attachCase(String theType,
-                                    String anEventId,
-                                    String exceptionRecordJurisdiction,
-                                    String caseTypeId,
+    private List<String> attachCase(String exceptionRecordJurisdiction,
                                     String caseRef,
-                                    List<Map<String, Object>> exceptionDocuments,
-                                    CaseDetails exceptionRecord) {
+                                    Long exceptionRecordReference,
+                                    List<Map<String, Object>> exceptionDocuments) {
         try {
-            attachCase(exceptionRecordJurisdiction, caseRef, exceptionRecord, exceptionDocuments);
+            doAttachCase(exceptionRecordJurisdiction, caseRef, exceptionDocuments, exceptionRecordReference);
             return success();
         } catch (CallbackException e) {
             return createErrorList(e);
         }
     }
 
-    //FOR the to do warnings
-    @SuppressWarnings("squid:S1135")
-    private void attachCase(String exceptionRecordJurisdiction,
-                            String caseRef,
-                            CaseDetails exceptionRecord,
-                            List<Map<String, Object>> exceptionDocuments) {
+    private void doAttachCase(String exceptionRecordJurisdiction,
```

Comment:
`attachCase` was good enough. but nvm ;)

Reply:
it's really to separate the error handling and the actual code. I personally find try catch statements clumsy and annoying and would rather pass back errors in a monad but ... well ... thats a little to FP for java.
If your interested have a look at the vavr Try.of().recover() stuff. 😄